libs/s25main/network/GameClient.cpp

```diff
@@ -1535,6 +1539,17 @@ bool GameClient::StartReplay(const boost::filesystem::path& path)
     return true;
 }
 
+void GameClient::SetAIBattlePlayers(std::vector<JoinPlayerInfo>&& playerInfos)
+{
+    aiBattlePlayers_ = playerInfos;
```

Comment:
The move needs to be explicit (which I admit is not intuitive, but see https://godbolt.org/z/j3zb3c5a3)

And then you can take it by value, as by R-value ref is unusual:
```suggestion
void GameClient::SetAIBattlePlayers(std::vector<JoinPlayerInfo> playerInfos)
{
    aiBattlePlayers_ = std:move(playerInfos);
```